### PR TITLE
Andrew doing/update on loaded

### PR DESCRIPTION
**Pull request by andrewDoing**
1. Updated onLoaded callback definition to accept GraphChatThread[], and return void (return not needed)
2. Removed conditional checking if a selected thread was already selected. This prevented a user from clicking on a thread, opening that chat, navigating back, and trying to reselect that same chat.
3. Updated samples to work

Closes # 

### PR checklist
- [ ] Project builds (`yarn build`) and changes have been tested in at least two supported browsers (Edge + non-Chromium based browser)
- [ ] All public APIs (classes, methods, etc) have been documented following the jsdoc syntax
- [ ] [Stories](https://github.com/microsoftgraph/microsoft-graph-toolkit/wiki/Storybook#writing-stories) have been added and existing stories have been tested
- [ ] Added appropriate [documentation](https://github.com/microsoftgraph/microsoft-graph-toolkit/wiki/Documentation). Docs PR: 

- [ ] License header has been added to all new source files (`yarn setLicense`)
- [ ] Contains **NO** breaking changes

### CODE REVIEW

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx [hunk 1]
```diff
@@ -188,21 +193,18 @@ export const ChatList = ({
   };
 
   const onClickChatListItem = (chatListItem: GraphChat) => {
-    // set selected state only once per click event
```
Comment (andrewDoing) [inline packages/mgt-chat/src/components/ChatList/ChatList.tsx:191]: below simply removes the top level if, to handle selecting the already selected chat

**Review by TechnicallyWilliams — APPROVED** [review]

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx [hunk 2]
```diff
@@ -178,7 +183,7 @@ export const ChatList = ({
       chatListClient.tearDown();
       setHeaderBannerMessage('We ran into a problem. Please close or refresh.');
     };
-  }, [chatListClient, onMessageReceived, onLoaded]);
+  }, [chatListClient, onMessageReceived, onLoaded, chatListState?.chatThreads]);
```
Comment (plasne) [inline packages/mgt-chat/src/components/ChatList/ChatList.tsx]: I think you can remove this now.

**Review by seekdavidlee — APPROVED** [review]

File: samples/react-contoso/src/pages/ChatPage.tsx [hunk 3]
```diff
@@ -112,11 +112,14 @@ const ChatPage: React.FunctionComponent = () => {
...
-      setChatId(e.id ?? '');
-    }
-  }, []);
+  const onChatSelected = React.useCallback(
+    (e: GraphChat) => {
+      if (chatId !== e.id) {
+        setChatId(e.id ?? '');
+      }
+    },
+    [chatId]
```
Comment (plasne) [inline samples/react-contoso/src/pages/ChatPage.tsx]: Does the function change because chatId changed?

**Review by plasne — CHANGES_REQUESTED** [review]
Please verify the comments I have.

**Review by plasne — APPROVED** [review]

**Pull request merged**